pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/search/ElasticSearch.java

```diff
@@ -156,6 +160,53 @@ String getDdpFromIndex(String searchHitIndex) {
         return searchHitIndex.substring(dotIndex + 1);
     }
 
+    public List<String> getAllParticipantsInIndex(String esParticipantsIndex) {
+        logger.info("Getting all participant ids from index " + esParticipantsIndex);
+        List<String> participantIds = new ArrayList<>();
+        SearchRequest searchRequest = new SearchRequest(esParticipantsIndex);
+        TimeValue scrollTimeout = TimeValue.timeValueMillis(100000L);
+        String scrollId = null;
+        try {
+            searchRequest.scroll(scrollTimeout);
+            SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+            searchSourceBuilder.query(QueryBuilders.matchAllQuery());
+            searchSourceBuilder.size(100);
+            searchSourceBuilder.sort("_doc", SortOrder.ASC); // Efficient scrolling
+            searchRequest.source(searchSourceBuilder);
+
+            SearchResponse searchResponse = ElasticSearchUtil.getClientInstance().search(searchRequest, RequestOptions.DEFAULT);
+            scrollId = searchResponse.getScrollId();
+            while (searchResponse.getHits().getHits().length > 0) {
+                for (SearchHit hit : searchResponse.getHits().getHits()) {
+                    participantIds.add(
+                            ((Map) hit.getSourceAsMap().get(ElasticSearchUtil.PROFILE)).get(ElasticSearchUtil.GUID).toString());
+                }
+                SearchScrollRequest scrollRequest = new SearchScrollRequest(scrollId);
+                scrollRequest.scroll(scrollTimeout);
+                searchResponse = ElasticSearchUtil.getClientInstance().scroll(scrollRequest, RequestOptions.DEFAULT);
+                scrollId = searchResponse.getScrollId();
+            }
+        } catch (IOException e) {
+            throw new DsmInternalError("Couldn't get participants from ES for instance " + esParticipantsIndex, e);
+        } finally {
+            if (scrollId != null) {
+                clearScroll(scrollId, esParticipantsIndex);
+            }
+        }
+
+        return participantIds;
+    }
+
+    private void clearScroll(String scrollId, String index) {
+        try {
+            ClearScrollRequest clearScrollRequest = new ClearScrollRequest();
+            clearScrollRequest.addScrollId(scrollId);
+            ElasticSearchUtil.getClientInstance().clearScroll(clearScrollRequest, RequestOptions.DEFAULT);
+        } catch (IOException e){
+            logger.error("Unable to clear the scroll in the connection to index: "+index, e);
```

Comment:
From what I understand, if the scroll is not cleared in Elasticsearch, it can lead to unnecessary resource usage on the Elasticsearch cluster. Each scroll request maintains a search context in the cluster, which consumes memory and other resources. If these contexts are not cleared, they will eventually time out based on the scroll timeout setting, but until they do, they can consume resources and potentially affect the performance of the Elasticsearch cluster, especially if there are many unclosed scrolls or if the cluster is under heavy load.